tools/gfx-unit-test/shader-cache-tests.cpp

```diff
@@ -689,6 +706,129 @@ namespace gfx_test
         }
     };
 
+    // Same as MultipleEntryShaderCache, but we now set the maximum entry count limit, so the cache
+    // should remove entries as needed when it reaches capacity.
+    //
+    // This test does not modify shaders as other tests already test this, instead focusing on checking
+    // that entries are correctly removed as cache limits are reached and that entries are always in
+    // the right order.
+    struct CacheWithMaxEntryLimit : MultipleEntryShaderCache
+    {
+        List<String> test1Lines; // C -> B
+        List<String> test2Lines; // A -> B
+        List<String> test3Lines; // A -> C
+        List<String> test4Lines; // C -> B -> A
+
+        void getCacheFile(List<String>& lines)
+        {
+            ComPtr<ISlangBlob> contentsBlob;
+            cacheFileSystem->loadFile(shaderCache.cacheFilename, contentsBlob.writeRef());
+            List<UnownedStringSlice> temp;
+            StringUtil::calcLines(UnownedStringSlice((char*)contentsBlob->getBufferPointer()), temp);
+            for (auto line : temp)
+            {
+                lines.add(line);
+            }
+        }
+
+        // Check the correctness of the cache's entries by comparing the order of entries in the
+        // current state of the cache with what we expect.
+        void checkCacheFiles()
+        {
+            // Check that shader A appears where we expect it to.
```

Comment:
Should also check the number of lines are the expected. 

Should check the corresponding files exist. 

Should check the supposed to be deleted file does not exist.

Reply:
Putting up a PR with these changes shortly